lib/charm/openstack/ha.py

```diff
@@ -0,0 +1,102 @@
+"""Classes for enabling ha in Openstack charms with the reactive framework"""
+
+import ipaddress
+
+import charmhelpers.contrib.network.ip as ip
+from charmhelpers.core.hookenv import unit_private_ip, config
+from relations.hacluster.common import CRM
+from relations.hacluster.common import ResourceDescriptor
+
+VIP_KEY = "vip"
+CIDR_KEY = "vip_cidr"
+IFACE_KEY = "vip_iface"
+
+"""Configure ha resources with:
+@when('ha.connected')
+def cluster_connected(hacluster):
+    ha.configure_ha_resources(hacluster, 'designate', ha_resources=['vips', 'haproxy'])
+
+TODO Proper docs to follow
+"""
+
+def configure_ha_resources(hacluster, service_name, ha_resources=None):
+    user_config = config()
+
+    RESOURCE_TYPES = {
+        'vips': configure_vips,
+        'haproxy': configure_haproxy,
+    }
+    if not ha_resources:
+        return
+    resources = CRM()
+    for res_type in ha_resources:
+        resources = RESOURCE_TYPES[res_type](resources, service_name)
+    hacluster.bind_on(iface=user_config['vip_iface'], mcastport=4440)
+    hacluster.manage_resources(resources)
+
+def configure_vips(_resources, service_name):
+    user_config = config()
+
+    for vip in user_config.get(VIP_KEY, []).split():
+        iface = (ip.get_iface_for_address(vip) or
+                 config(IFACE_KEY)) 
+        netmask = (ip.get_netmask_for_address(vip) or
+                   config(CIDR_KEY))
+        if iface is not None:
+            _resources.add(
+                VirtualIP(
+                    service_name,
+                    vip,
+                    nic=iface,
+                    cidr=netmask
+                )
```

Comment:
Minor style point, but generally it's prefered to not have 'dangling' braces.  e.g. `cidr=netmask,))`